lib/dxf/entity.rb

```diff
@@ -147,7 +147,22 @@ def initialize(*points)
 
     # Return the individual line segments
     def lines
-      points.each_cons(2).map { |a, b| Line.new a, b }
+      points.each_cons(2).map do |a, b|
+        line = Line.new a,b
+        line.layer = self.layer
+        line.color_number = self.color_number
+        line.x1 = a[0]
+        line.y1 = a[1]
+        line.x2 = b[0]
+        line.y2 = b[1]
+        line
+      end
+    end
+  end
+
+  class Polyline < LWPolyline
+    def initialize(*points)
+      @points = points.drop(1).map {|a| Point[a]}
```

Comment:
the Polyline and LWPolyline share properties, the main difference is that the first is formed by VERTEX entities, the first point on the Polyline is always {0,0} and can be ignored

https://documentation.help/AutoCAD-DXF/WS1a9193826455f5ff18cb41610ec0a2e719-79f0.htm